substratevm/src/com.oracle.svm.agent/src/com/oracle/svm/agent/BreakpointInterceptor.java

```diff
@@ -1241,6 +1279,9 @@ public static void onUnload() {
                                     "(Ljava/lang/ClassLoader;[Ljava/lang/Class;Ljava/lang/reflect/InvocationHandler;)Ljava/lang/Object;", BreakpointInterceptor::newProxyInstance),
 
                     brk("java/io/ObjectStreamClass", "<init>", "(Ljava/lang/Class;)V", BreakpointInterceptor::objectStreamClassConstructor),
+                    brk("jdk/internal/reflect/ReflectionFactory",
```

Comment:
On Java 8, this class would be called `sun.reflect.ReflectionFactory` - since this breakpoint is not optional, this would make the agent always fail on Java 8 as it won't be able to resolve this class. We could either make this breakpoint an `optionalBrk` or we could generate the class name based on the JDK version (using `com.oracle.svm.core.jdk.Package_jdk_internal_reflect#getQualifiedName`)

Reply:
Nice catch! Done!